utilgo/tlsmanager/tlsmanager.go

```diff
@@ -203,12 +213,21 @@ func (m *tlsManager) WatchClientTLSConf(clientName ClientName, handler func(conf
 		case kclient.SecretChangeTypeAdd:
 			fallthrough
 		case kclient.SecretChangeTypeUpdate:
+			if m.tlsClientCertPEM[clientName] != nil && bytes.Compare(secret.Data[TLSSecretCrtName], m.tlsClientCertPEM[clientName]) == 0 {
+				return
+			}
+
+			m.tlsClientCertPEM[clientName] = secret.Data[TLSSecretCrtName]
 			rootCA := m.BestEffortSystemCertPool()
 			rootCA.AppendCertsFromPEM(secret.Data[TLSSecretCrtName])
 			conf := &tls.Config{RootCAs: rootCA}
 			m.tlsClientConfigs[clientName] = conf
 			handler(conf, nil)
 		case kclient.SecretChangeTypeDelete:
+			if m.tlsClientCertPEM[clientName] == nil {
```

Comment:
Shouldn't this be m.tlsClientConfigs instead?

Reply:
I think it's better to rely on `tlsClientCertPEM` map as on source of truth and always check those values. So below I've added `delete(m.tlsClientCertPEM, clientName)` line to also delete cert pem.